From 715277cfadd7b3872ad6f7001fe84fb7568c947c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gonzalo=20Tornar=C3=ADa?= <tornaria@cmat.edu.uy>
Date: Mon, 13 Dec 2021 18:04:04 -0300
Subject: update ipykernel and make debugpy optional

 - update ipykernel to 6.6.0, for which debugpy is optional
 - patch out debugpy from pyproject.toml and setup.py
 - make debugpy optional for sagemath
---
 build/pkgs/debugpy/type                             |  2 +-
 build/pkgs/ipykernel/checksums.ini                  |  6 +++---
 build/pkgs/ipykernel/dependencies                   |  2 +-
 build/pkgs/ipykernel/package-version.txt            |  2 +-
 .../ipykernel/patches/debugpy-make-optional.patch   | 21 +++++++++++++++++++++
 5 files changed, 27 insertions(+), 6 deletions(-)
 create mode 100644 build/pkgs/ipykernel/patches/debugpy-make-optional.patch

diff --git a/build/pkgs/debugpy/type b/build/pkgs/debugpy/type
index a6a7b9c..134d9bc 100644
--- a/build/pkgs/debugpy/type
+++ b/build/pkgs/debugpy/type
@@ -1 +1 @@
-standard
+optional
diff --git a/build/pkgs/ipykernel/checksums.ini b/build/pkgs/ipykernel/checksums.ini
index ca1302e..724eec8 100644
--- a/build/pkgs/ipykernel/checksums.ini
+++ b/build/pkgs/ipykernel/checksums.ini
@@ -1,5 +1,5 @@
 tarball=ipykernel-VERSION.tar.gz
-sha1=2c2aa0e929bfd0e694da6b45d2de846a0a5d2812
-md5=78797cb2389d8f131c75280f808aac15
-cksum=1454395629
+sha1=5a5cf7f8c0c02d0c0cc5fe3e0fe7481a86de6552
+md5=f940975eb00de793695c386ad3a8800c
+cksum=597841676
 upstream_url=https://pypi.io/packages/source/i/ipykernel/ipykernel-VERSION.tar.gz
diff --git a/build/pkgs/ipykernel/dependencies b/build/pkgs/ipykernel/dependencies
index eec7126..5b3708a 100644
--- a/build/pkgs/ipykernel/dependencies
+++ b/build/pkgs/ipykernel/dependencies
@@ -1,4 +1,4 @@
-$(PYTHON) ipython_genutils importlib_metadata argcomplete debugpy matplotlib_inline ipython jupyter_client tornado appnope traitlets | $(PYTHON_TOOLCHAIN)
+$(PYTHON) ipython_genutils importlib_metadata argcomplete matplotlib_inline ipython jupyter_client tornado appnope traitlets | $(PYTHON_TOOLCHAIN)
 
 ----------
 All lines of this file are ignored except the first.
diff --git a/build/pkgs/ipykernel/package-version.txt b/build/pkgs/ipykernel/package-version.txt
index a194c18..826f5ce 100644
--- a/build/pkgs/ipykernel/package-version.txt
+++ b/build/pkgs/ipykernel/package-version.txt
@@ -1 +1 @@
-6.5.1
+6.6.0
diff --git a/build/pkgs/ipykernel/patches/debugpy-make-optional.patch b/build/pkgs/ipykernel/patches/debugpy-make-optional.patch
new file mode 100644
index 00000000..308e4dd
--- /dev/null
+++ b/build/pkgs/ipykernel/patches/debugpy-make-optional.patch
@@ -0,0 +1,21 @@
+--- a/pyproject.toml	2021-12-01 10:23:30.000000000 -0300
++++ b/pyproject.toml	2021-12-13 18:01:36.239921204 -0300
+@@ -3,7 +3,6 @@
+ requires=[
+   "setuptools",
+   "wheel",
+-  "debugpy",
+   "ipython>=5",
+   "jupyter_core>=4.2",
+   "jupyter_client",
+diff -ruN a/setup.py b/setup.py
+--- a/setup.py	2021-12-01 10:23:09.000000000 -0300
++++ b/setup.py	2021-12-13 18:01:40.112873823 -0300
+@@ -63,7 +63,6 @@
+     install_requires=[
+         'importlib-metadata<5;python_version<"3.8.0"',
+         'argcomplete>=1.12.3;python_version<"3.8.0"',
+-        'debugpy>=1.0.0,<2.0',
+         'ipython>=7.23.1',
+         'traitlets>=5.1.0,<6.0',
+         'jupyter_client<8.0',
-- 
cgit v1.0-1-gd88e

